File: frontend/ecommerce-react/src/types/product.ts

export type ProductImage = {
  id: number;
  url: string;
};

export type Product = {
  id: number;
  name: string;
  price: number;
  description: string;
  stock: number;
  statusLabel?: string;
  categoryId?: string;
  images: ProductImage[];
};

export type ProductStatus = "active" | "inactive";

export const ProductStatusValues: Record<string, ProductStatus> = {
  Active: "active",
  Inactive: "inactive",
};


export interface ProductFilters {
  CategoryId?: number;
  PriceMin?: number;
  PriceMax?: number;
  Status?: string; 
}

export interface SearchParams {
  search: string;
  priceMin?: number;
  priceMax?: number;
}

File: backend/ecommerce-net/src/core/domain/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Net.Mime;
using domain.Common;

namespace domain;

public class Product : BaseDomainModel
{
    [StringLength(100)]
    [Column(TypeName = "NVARCHAR(100)")]
    public string? Name { get; set; }
    [Column(TypeName = "NVARCHAR(4000)")]
    public string? Description { get; set; }
    [Column(TypeName = "DECIMAL(10,2)")]
    public decimal Price { get; set; }

    public int Stock { get; set; }
    public ProductStatus Status { get; set; } = ProductStatus.Activo;
    
    public int CategoryId { get; set; }
    public Category? Category { get; set; }
    
    public virtual ICollection<Image>? Images { get; set; }
    
}